WcaOnRails/app/models/competition.rb

```diff
@@ -222,6 +235,7 @@ def update_foreign_keys_when_id_changes
       CompetitionMedium.where(competitionId: id_was).update_all(competitionId: id)
       CompetitionDelegate.where(competition_id: id_was).update_all(competition_id: id)
       CompetitionOrganizer.where(competition_id: id_was).update_all(competition_id: id)
+      DelegateReport.where(competition_id: id_was).update_all(competition_id: id)
```

Comment:
@jonatanklosko, that cleanup sounds good, except I split the list of models into multiple lines for ease of diff reading and conflict resolution.

@larspetrus, can you explain the test you have in mind a bit more closely? Following your idea from a previous PR, I was thinking about a list of `MODELS_WITH_COMPETITION_ID` and `MODELS_WITHOUT_COMPETITION_ID`, and then we assert that the list of tables for our models is the same as the set of tables in the database?